### PR TITLE
Update who must approve repo creation requests

### PR DESCRIPTION
**Pull request by nikhita**
Current state of docs:

- https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md#subproject-creation
mentions that repos _may_ be created under k-sigs by a lazy consensus of subproject owners.

- Some charters like https://github.com/kubernetes/community/blob/master/sig-api-machinery/charter.md#deviations-from-sig-governance
have explicit processes setup for who can approve repo creation requests.

- https://github.com/kubernetes/community/blob/master/github-management/kubernetes-repositories.md#rules-for-new-repositories
says that repo creation requests must be approved by lazy consensus of SIG membership.

Current state of how we do things:

- We create repos if the SIG leads have approved the repo creation
request.
- We don't require a lazy consensus thread on the mailing list.

To reconcile both docs and the current state of things, this commit
updates the docs to say that repo creation requests must be approved by
lazy consensus of SIG membership _and_ must also be explicitly approved
by folks listed out the in the SIG charter.

Note that the wording says "process spelled out in the SIG charter"
because SIG charters _may_ also define a complete process on how they'd
like to handle this, not just the list of people who can approve repo
creation requests.

/assign @spiffxp @cblecker

### CODE REVIEW

**Comment by nikhita** [comment]
> Subprojects may create repos under github.com/kubernetes-sigs through lazy-consensus of subproject owners.

One thing to note here is that, unless:

- A SIG has explicitly specified who can approve repo creation requests in their charter (like [apimachinery](https://github.com/kubernetes/community/blob/master/sig-api-machinery/charter.md#deviations-from-sig-governance))
- A subproject consists of a single repo, whereby subproject owners = approvers in that repo

it's usually unclear who the subproject owners really are and so we generally default to approval by SIG leads.

**Comment by nikhita** [comment]
TLDR: the only thing that is being changed from how we handle things right now is enforcing lazy consensus. I'll update the issue template in k/org once this gets lgtm.

**Comment by spiffxp** [comment]
I agree that in practice I've almost never seen the subproject stuff in charter template followed to the letter.  That said I'm not sure that the spirit is lazy-consensus everything.  I'm much more interested in enforcing that there is a publicly linkable written decision.

We can try this to see that it works, but as you've pointed out, there's cruft and wiggle room everywhere.  [Tightening up who subproject owners are](https://github.com/kubernetes/community/issues/1673#issuecomment-484746226) may help if we need to revisit this.

**Comment by nikhita** [comment]
> I'm much more interested in enforcing that there is a publicly linkable written decision.

Sgtm.

**Comment by cblecker** [comment]
I don't think that the repo creation guidelines are the right place for this. I'd rather something like "approval must follow subproject creation guidance per the sig's charter" and just remove the extra language in our doc.

Basically, I want to leave it to the sigs to tell us if something is approved, and not have us (GH admin) be the police. Follow your charter, someone from leadership should sign off, then we do the thing.

**Comment by nikhita** [comment]
Updated. PTAL.

Leaving the lazy consensus part for repo creation in https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md#subproject-creation as-is because it says that SIGs _MAY_ do that (not _MUST_).

> I'd rather something like "approval must follow subproject creation guidance per the sig's charter"

I updated to just "process spelled out in the SIG's charter" because SIGs may define approval for repos created in existing subprojects differently than creating repos for new subprojects, example: https://github.com/kubernetes/community/blob/master/sig-api-machinery/charter.md#deviations-from-sig-governance.

**Comment by cblecker** [comment]
/lgtm
/approve

Thanks @nikhita
